### DISCUSSION TITLE
Is there a specific status for strongswan to build a tunnel?

### DISCUSSION THREAD

**Discussion by XYQ202** [strongswan/strongswan#1173] [Q&A]
hello，team

I want to get the status of the tunnel setup between the device and the gateway, such as the status of **the first stage of negotiation**, **the second stage of negotiation**, the status of **establishment or failure**, etc. I would like to ask if you can return different statuses according to different flags.

Because I can't view it through the command line, such as ipsec status, etc.

**Comment by tobiasbrunner** [comment]
You might want to look into [vici](https://docs.strongswan.org/docs/5.9/plugins/vici.html) (and perhaps the [_error-notify_ plugin](https://docs.strongswan.org/docs/5.9/plugins/error-notify.html)).

**Reply by XYQ202** [reply]
Thanks for your reply, I installed the error-notify plugin, but how should I use it, can you provide more details？

**Reply by tobiasbrunner** [reply]
The documentation linked above has more information on the socket interface.
